web/src/lib/components/photos-page/asset-grid.svelte

```diff
@@ -459,6 +459,18 @@
     }
   };
 
+  const downloadSelectedAssets = async () => {
+    if ($selectedAssets.size === 1) {
+      const [asset] = $selectedAssets;
+      assetInteractionStore.clearMultiselect();
+      await downloadFile(asset);
+      return;
+    }
+
+    assetInteractionStore.clearMultiselect();
```

Comment:
We should probably clear the selection _after_ the download happens. Also, can you wrap this in a try/catch that uses `handleError` to send a notification to the user if something goes wrong.

Reply:
The `downloadFile` promise only resolves once the file is fully downloaded. Clearing the selection afterwards could be problematic, because the user can change the selection in the meantime. This function already does some error handling and will show a notification if the download fails, so I'm not sure if the try/catch is even needed.

Reply:
I see.